XenAdmin/SettingsPanels/NRPEEditPage.cs

```diff
@@ -0,0 +1,516 @@
+                    !REGEX_IPV4_CIDR.Match(hostArray[i].Trim()).Success &&
+                    !REGEX_DOMAIN.Match(hostArray[i].Trim()).Success)
+                {
+                    _invalidParamToolTipText = Messages.NRPE_ALLOW_HOSTS_FORMAT_ERROR;
+                    return false;
+                }
+                for ( int j = 0; j < i; j++ )
+                {
+                    if (hostArray[i].Trim().Equals(hostArray[j].Trim()))
+                    {
+                        _invalidParamToolTipText = Messages.NRPE_ALLOW_HOSTS_SAME_ADDRESS;
+                        return false;
+                    }
+                }
+            }
+            foreach (string host in hostArray)
+            {
+                if (!REGEX_IPV4.Match(host.Trim()).Success &&
+                    !REGEX_IPV4_CIDR.Match(host.Trim()).Success &&
+                    !REGEX_DOMAIN.Match(host.Trim()).Success)
+                {
+                    _invalidParamToolTipText = Messages.NRPE_ALLOW_HOSTS_FORMAT_ERROR;
+                    return false;
+                }
+            }
+            CheckDataGridView.ShowCellToolTips = false;
+            return true;
+        }
+
+        private string AllowHostsWithoutLocalAddress(string allowHosts)
+        {
+            string UpdatedAllowHosts = "";
+            string[] AllowHostArray = allowHosts.Split(',');
+            foreach (string allowHost in AllowHostArray)
+            {
+                if (!allowHost.Trim().Equals("127.0.0.1") &&
+                    !allowHost.Trim().Equals("::1"))
+                {
+                    UpdatedAllowHosts += allowHost + ",";
+                }
+            }
+            return UpdatedAllowHosts.Length == 0 ? NRPEHostConfiguration.ALLOW_HOSTS_PLACE_HOLDER :
+                UpdatedAllowHosts.Substring(0, UpdatedAllowHosts.Length - 1);
+        }
+
+        private bool IsNRPEConfigurationChanged()
+        {
+            if (_nrpeCurrentConfig.EnableNRPE != _nrpeOriginalConfig.EnableNRPE ||
+                !_nrpeCurrentConfig.AllowHosts.Equals(_nrpeOriginalConfig.AllowHosts) ||
+                _nrpeCurrentConfig.Debug != _nrpeOriginalConfig.Debug ||
+                _nrpeCurrentConfig.SslLogging != _nrpeOriginalConfig.SslLogging)
+            {
+                return true;
+            }
+            foreach (KeyValuePair<string, NRPEHostConfiguration.Check> kvp in _nrpeCurrentConfig.CheckDict)
+            {
+                NRPEHostConfiguration.Check CurrentCheck = kvp.Value;
+                _nrpeOriginalConfig.GetNRPECheck(kvp.Key, out NRPEHostConfiguration.Check OriginalCheck);
+                if (CurrentCheck != null && OriginalCheck != null
+                    && (!CurrentCheck.WarningThreshold.Equals(OriginalCheck.WarningThreshold)
+                    || !CurrentCheck.CriticalThreshold.Equals(OriginalCheck.CriticalThreshold)))
+                {
+                    return true;
+                }
+            }
+            return false;

```

Comment:
I'd put this logic in the `Equals` override for `NRPEHostConfiguration`, so they can be reused. then you can just use a nice and simple `Equals` here.